modules/ensemble/lib/widget/inline_timepicker.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/box_wrapper.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:ensemble/framework/action.dart' as ensemble;
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/action/haptic_action.dart';
+import 'package:intl/intl.dart'; // For date formatting
+import '../framework/event.dart';
+
+/// An inline time picker widget that mimics the iOS clock app's circling time picker.
+/// It provides various customization options for a seamless user experience.
+class InlineTimePicker extends StatefulWidget
+    with
+        Invokable,
+        HasController<InlineTimePickerController, InlineTimePickerState> {
+  static const type = 'InlineTimePicker';
+
+  InlineTimePicker({Key? key}) : super(key: key);
+
+  final InlineTimePickerController _controller = InlineTimePickerController();
+
+  @override
+  get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => InlineTimePickerState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'selectedTime': () => _controller.selectedTimeFormatted(),
+    };
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {
+      'reset': () => _controller.reset(),
+    };
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      'initialTime': (time) =>
+          _controller.initialTime = Utils.getTimeOfDay(time),
+      'onTimeChanged': (funcDefinition) => _controller.onTimeChangedAction =
+          ensemble.EnsembleAction.from(funcDefinition, initiator: this),
+      'mode': (value) => _controller.mode =
+          Utils.getEnum<CupertinoTimerPickerMode>(
+              value, CupertinoTimerPickerMode.values),
+      'minuteInterval': (value) =>
+          _controller.minuteInterval = Utils.optionalInt(value),
+      'secondInterval': (value) =>
+          _controller.secondInterval = Utils.optionalInt(value),
+      'onTimeChangedHaptic': (value) =>
+          _controller.onTimeChangedHaptic = Utils.optionalString(value),
+    };
+  }
+}
+
+class InlineTimePickerController extends BoxController {
```

Comment:
see the comment right above BoxController - ` TODO: Legacy - move to EnsembleBoxController`

use `EnsembleBoxController`